docs/project-development-environment.md

```diff
@@ -2,7 +2,19 @@
+| <a name='dependency_python3'>Python 3</a> | 3.4 | Would be pulled in by `python3-gdal` or <a href='#dependency_python3-dev'>`python3-dev`</a>, but you should install it explicitly with <pre class="highlight highlight-source-shell">sudo apt install python3</pre> |
+| <a name='dependency_requirements.txt'>various PyPI packages</a> | See [`requirements.txt`](/requirements.txt) | Install in a Python 3 [virtual environment](#dependency_venv). Create one with e.g. <pre class="highlight highlight-source-shell">mkdir -p ~/.virtualenvs && virtualenv ~/.virtualenvs/osmaxx -p python3</pre> activate it with <pre class="highlight highlight-source-shell">source ~/.virtualenvs/osmaxx/bin/activate</pre> Then, in the same shell session, use [`pip3`](#dependency_pip3) to install the packages: <pre class="highlight highlight-source-shell">pip3 install -r requirements.txt  # run from this repo's root dir</pre><hr>`virtualenvwrapper` users can perform all of the above in a single step: <pre class="highlight highlight-source-shell">mkvirtualenv -a . -r requirements.txt -p python3 osmaxx  # run from this repo's root dir</pre>The `-a .` will also [associate the repo root as the project directory](http://virtualenvwrapper.readthedocs.org/en/latest/command_ref.html#mkvirtualenv) of the new Python 3 virtual environemnt. |
+| <a name='dependency_pip3'>pip3</a> |  | Provided by <a href='#dependency_venv'>virtualenv</a> in the virtual Python environments it creates. |
+| <a name='dependency_venv'>virtualenv</a> (Python 2 and 3) |  | Install system-wide with <pre class="highlight highlight-source-shell">sudo apt install python-virtualenv</pre> |
+| <a name='dependency_python3-dev'>Python 3 C bindings</a> | 3.4 | Would be pulled in by `python3-gdal`, but you should install it explicitly with <pre class="highlight highlight-source-shell">sudo apt install python3-dev</pre> | 
+| <a name='dependency_geos_c'>GEOS C library</a> | whatever [GeoDjango supports](https://docs.djangoproject.com/en/1.9/ref/contrib/gis/install/geolibs/#installing-geospatial-libraries) | <pre class="highlight highlight-source-shell">sudo apt install python3-gdal</pre> will pull this and other required libraries in. `python3-gdal` itself is not required. Thus, if you prefer a more minimal installation, only install `libgeos-c1`. |
+| <a name='dependency_gdal'>GDAL library</a> | whatever [GeoDjango supports](https://docs.djangoproject.com/en/1.9/ref/contrib/gis/install/geolibs/#installing-geospatial-libraries) | <pre class="highlight highlight-source-shell">sudo apt install python3-gdal</pre> will pull this and other required libraries in. `python3-gdal` itself is not required. Thus, if you prefer a more minimal installation, only install `libgdal1h`. |
+| <a name='dependency_pip'>pip</a> (Python 2) |  | <pre class="highlight highlight-source-shell">sudo apt install python-pip</pre> |
```

Comment:
For some elements, the choice of how to install them affects what other elements need to be installed first. Also depending on how you install the former, the latter might be pulled in automatically by the installation system or not. Thus, we would have to list some optional elements before some mandatory elements with the decision whether you want an optional one or not depends on how you want to install a later listed mandatory one. That would be rather confusing, wouldn’t it?

Therefore, I tried to list them in the order you have to make your decisions. Not yet sure whether that makes more sense, though.